### ISSUE TITLE
[Feature]: Add an IsDestroyed method on WebView2 that tells us if the process behind the control is still around

### ISSUE THREAD

**Issue by pushkin-** [MicrosoftEdge/WebView2Feedback#4342]
### Describe the feature/enhancement you need

Electron has an [`isDestroyed()`](https://www.electronjs.org/docs/latest/api/web-contents#contentsisdestroyed) that we can call on a webcontents/browserview to check if the process is still around. I'm not seeing a similar API/property in Webview2

I can intercept `WindowCloseRequested ` or better yet `ProcessFailed` and set some flag saying it's destroyed (though maybe realistically I would just exit the application in this situation), but a property/method check would be nice.

### The scenario/use case where you would use this feature

Just extra safety. We have a webview2 control property on the main Form. Would be nice if we can ask if it's still around before donig stuff with it (the BrowserProcessId property still shows the old PID even if that process is now destroyed)

### How important is this request to you?

Nice to have. There are other ways to tackle this, but having official API support would be beneficial.

### Suggested implementation

Some kind of `IsDestroyed` property or method would be nice.

[AB#48857882](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/48857882)

**Comment by victorhuangwq** [comment]
@LiangTheDev do you know if what he is trying to achieve is currently possible?

**Comment by LiangTheDev** [comment]
I think that this request is for adding a property like `Closed` on WebView objects. Before we add the property, a workaround would be add a helper to try to access some existing property on WebView2Controller (like ZoomFactor) or WebView2 (like CanGoBack) and see whether it succeeds or throws.

Yes, different process failures are handled differently. WebView2 would try to auto recovery from certain failures, let GPU or other utility process failures, while main WebView2 process failure would cause WebView2 to be closed and renderer process failure would be reported to host with error page shown in WebView2.
Whether we add Closed property or not, the app should handle ProcessFailed event to handle different types of the failures.

**Comment by LiangTheDev** [comment]
@victorhuangwq I don't think that the property currently exist. While there are workarounds and the best practice is to listen to ProcessFailed event, it might be beneficial to have an easy way to check whether the WebView2 control can still be used. Currently, things are not clear, even accessing CoreWebView2 property could throw in WPF WebView2 control.

**Comment by victorhuangwq** [comment]
Thanks @LiangTheDev. I will track this feature request in our backlog.

**Comment by champnic** [comment]
✅ Successfully linked to Azure Boards work item(s):
- [Scenario 48857882](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/48857882)
